platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultDevelocityBuildLifecycleService.java

```diff
@@ -27,14 +28,27 @@
 public class DefaultDevelocityBuildLifecycleService implements DevelocityBuildLifecycleService {
 
     private final Gradle gradle;
+    private final BuildFeatures buildFeatures;
 
     @Inject
-    public DefaultDevelocityBuildLifecycleService(Gradle gradle) {
+    public DefaultDevelocityBuildLifecycleService(Gradle gradle, BuildFeatures buildFeatures) {
         this.gradle = gradle;
+        this.buildFeatures = buildFeatures;
     }
 
     @Override
     public void beforeProject(Action<? super Project> action) {
-        gradle.getLifecycle().beforeProject(action::execute);
```

Comment:
I disagree here. The `beforeProject` action passed by the scan plugin is isolated projects compatible. The problem comes from the test JVM possibly missing the required `add-opens` statements to isolate Lambdas. An actual Gradle daemon JVM will always have the correct `add-opens` statements.

Reply:
Can the `beforeProject` action passed by the plugin contain any references to user data or user-defined actions?

Reply:
No, it only runs code owned by the plugin.

Reply:
That makes sense. FTR: I agreed with @wolfs that we would improve the message in TestKit to make it more actionable in 8.11 and revert this change.